webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -232,6 +247,18 @@ def update_response_overview_plot(
         fig.update_layout(clickmode="event+select")
         return fig
 
+    def get_first_observation_x(obs_data: pd.DataFrame) -> Any:
+        """
+        :return: The first x value in the observation data, converted
+            to type suitable for lookup in the response vector.
+        """
+        if type(obs_data["x_axis"][0]) == str:
+            return int(obs_data["x_axis"][0])
+        elif type(obs_data["x_axis"][0]) == pd._libs.tslibs.timestamps.Timestamp:
+            return str(obs_data["x_axis"][0])
+        else:
+            return obs_data["x_axis"][0]
+
```

Comment:
I am not comfortable with the need of this function. 
Can we try to get rid of the inconsistence between data type of the x axis from the observation and response?
Maybe it is something necessary or even a feature?

Reply:
I see, so it is not inconsistency if we need both. Eventually I will understand why :D

Reply:
Perhaps prefix function with _get.. if its internal to this class and do a Union of the possible return types

Reply:
That could be a good idea if its an error if the datatype is not str or timestamp.

Reply:
ok, a note is that `get_first_observation_x` is part of the function `response_correlation_controller` and not part of a class. But I guess it is still valid your suggestion.